internal/server/middleware.go

```diff
@@ -49,63 +50,80 @@ func DatabaseMiddleware(db *gorm.DB) gin.HandlerFunc {
 	}
 }
 
-func DestinationMiddleware() gin.HandlerFunc {
-	return func(c *gin.Context) {
-		uniqueID := c.GetHeader("Infra-Destination")
-		if uniqueID != "" {
-			destinations, err := access.ListDestinations(c, uniqueID, "", &models.Pagination{})
-			if err != nil {
-				return
-			}
+func updateDestinationLastSeenAt(c *gin.Context) error {
+	uniqueID := c.Request.Header.Get("Infra-Destination")
+	if uniqueID == "" {
+		return nil
+	}
 
-			switch len(destinations) {
-			case 0:
-				// destination does not exist yet, noop
-			case 1:
-				destination := destinations[0]
-				// only save if there's significant difference between LastSeenAt and Now
-				if time.Since(destination.LastSeenAt) > time.Second {
-					destination.LastSeenAt = time.Now()
-					if err := access.SaveDestination(c, &destination); err != nil {
-						sendAPIError(c, err)
-						return
-					}
-				}
-			default:
-				sendAPIError(c, fmt.Errorf("multiple destinations found for unique ID %q", uniqueID))
-				return
+	// TODO: use GetDestination(ByUniqueID())
+	destinations, err := access.ListDestinations(c, uniqueID, "", &models.Pagination{})
+	if err != nil {
+		return err
```

Comment:
Previously we would ignore this error, which is probably not a huge problem, but it seemed more correct to return the error now.